vscode/src/completions/vscodeInlineCompletionItemProvider.ts

```
@@ -295,6 +309,44 @@ export class InlineCompletionItemProvider implements vscode.InlineCompletionItem
             })
         })
     }
+
+    /**
+     * A callback that is called whenever an error happens. We do not want to flood a users UI with
+     * error messages so every unexpected error is deduplicated by its message and rate limit errors
+     * are only shown once during the rate limit period.
+     */
+    private onError(error: Error | RateLimitError): void {
+        if (error instanceof RateLimitError) {
+            if (this.resetRateLimitErrorsAfter && this.resetRateLimitErrorsAfter > Date.now()) {
+                return
+            }
+            this.resetRateLimitErrorsAfter = error.retryAfter?.getTime() ?? Date.now() + 24 * 60 * 60 * 1000
+            this.config.statusBar.addError({
+                title: 'Cody Autocomplete Disabled Due to Rate Limit',
+                description:
+                    `You've used all${error.limit ? ` ${error.limit}` : ''} daily autocompletions.` +
+                    (error.retryAfter ? ` Usage will reset in ${formatDistance(error.retryAfter, new Date())}.` : ''),
+                onSelect: () => {
+                    void vscode.env.openExternal(
+                        vscode.Uri.parse('https://about.sourcegraph.com/blog/increasing-the-completions-rate-limit')
+                    )
+                },
+            })
+            return
+        }
+
+        if (this.reportedErrorMessages.has(error.message)) {
+            return
+        }
+        this.reportedErrorMessages.add(error.message)
```

Does this mean that non-rate-limit errors will be shown only once per the activated extension lifetime? 

It would be helpful to show all the errors (even if a user already saw errors with this particular message) if they happen after a certain period. This way, we ensure we avoid flooding them with the same error messages and keep users informed about new errors occurring without swallowing them.